### PR TITLE
IirFilterNode implementation

### PR DESCRIPTION
**Pull request by Jerboas86**
Implementation of the IirFilterNode.

### Done
- Api (supposedly)  Compliant with the web audio api specs
- Add an example iir.rs
- Add docs

### Todos
- [x] Optimizations
- [x] Add tests

related to #23

### CODE REVIEW

File: src/node/iir_filter.rs [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    /// # Arguments
+    ///
+    /// * `input` - Audiobuffer input
+    #[inline]
+    fn tick(&mut self, input: f32) -> f32 {
+        let input = input as f64;
+        let output = self.norm_coeffs[0].0 * input + self.states[0];
+
+        for (idx, (ff, fb)) in self.norm_coeffs.iter().skip(1).enumerate() {
+            self.states[idx] = ff * input - fb * output + self.states.get(idx + 1).unwrap_or(&0.);
```
[the remark this answers is not in the document]
Comment (Jerboas86) [reply src/node/iir_filter.rs]: I think i found another solution. 
`states` length is now the same as `norm_coeffs` length.
this extra state is only read and is always 0

**Comment by Jerboas86** [comment]
i will replace white noise with sine signals that should make it work.

File: src/node/iir_filter.rs [hunk 2]
```diff
@@ -0,0 +1,652 @@
...
+        background.connect(&iir);
+        iir.connect(&context.destination());
+
+        background.start();
+        let output = context.start_rendering();
+
+        // retrieve processed data by removing silence chunk
+        // These silence slices are inserted inconsistently from an test execution to another
+        // Without these processing the test would be brittle
+        let data_ch: Vec<f32> = output
```
Comment (Jerboas86) [inline src/node/iir_filter.rs:638]: I retry the test a bunch of time with a constant success.

Comment (orottier) [reply src/node/iir_filter.rs:638]: I think the inconsistencies come from the fact that you have used a MediaElement, which will perform some buffering that can be non-deterministic in tests. Can you try to use `create_media_stream_source` on the stream? That fixes the problem I hope and gives a bit more confidence in the test

Comment (Jerboas86) [reply src/node/iir_filter.rs:638]: i will try your proposed alternative tomorrow

Comment (Jerboas86) [reply src/node/iir_filter.rs:638]: > Can you try to use create_media_stream_source on the stream?

I tried with `create_media_stream_source` and it also fails from time to time if i don't add the counter measures of the actual test code

Comment (orottier) [reply src/node/iir_filter.rs:638]: That just sounds really strange to me.... But probably unrelated to your work. Let's get it merged

**Pull request merged**